### PR TITLE
add lame 3.100

### PR DESCRIPTION
**Pull request by soumith**
Added LAME mp3 encoder. Once this is in, I'll update the `sox` feedstock to include LAME in it's build, so that we can encode / decode mp3 using it.

cc: @conda-forge/help-c-cpp

### CODE REVIEW

File: recipes/lame/build.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+export CFLAGS="-I$PREFIX/include $CFLAGS"
```
Comment (soumith) [inline recipes/lame/build.sh]: the first 15 lines and the last 2 lines have been lifted from the ImageMagick feedstock's `build.sh`

**Comment by soumith** [comment]
just FYI, I used the patch and commands in the Homebrew Formula to get the job done. For reference: https://github.com/Homebrew/homebrew-core/blob/master/Formula/lame.rb

File: recipes/lame/build.sh (hunk 1 above)
Comment (jakirkham) [reply recipes/lame/build.sh]: Yeah the ImageMagick feedstocks is a bit outdated it seems then. These flags are included in our legacy and new toolchains. So all of these can be dropped in the feedstock. This includes the `if`/`elif` below as well.

Also the `.la` (libtool archive) file removal is handled by `conda-build` directly. The fixes for this below preceded that release and can be dropped too.

Hopefully this simplifies things for you a bit going forward. :)

File: recipes/lame/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  patches:
+    - libmp3lame.patch
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (jakirkham) [inline recipes/lame/meta.yaml]: In some cases, we have needed to add an empty `host` section as well to ensure the compilers and other build tools are placed in a separate environment from the install prefix.

File: recipes/lame/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+test:
+  commands:
+    - lame --genre-list testcase.mp3
+
+about:
+  home: https://lame.sourceforge.io/
+  license: LGPL
```
Comment (jakirkham) [inline recipes/lame/meta.yaml]: Generally we try to include license version here as well (e.g. LGPL 2.1 or LGPL 3 depending, have not checked myself for LAME specifically).

**Comment by jakirkham** [comment]
Thanks @soumith. This is a nice addition to have.

There are a few other audio/video libraries that we might want to rebuild to use `lame` now that it is present.

Also added a few comments above, which can be addressed in the feedstock. Hopefully that simplifies the maintenance effort a bit going forward.

**Pull request merged**